.cypress/integration/ad/dashboard/ad_dashboard.spec.ts

```diff
@@ -13,31 +13,67 @@
  * permissions and limitations under the License.
  */
 
-/// <reference types="cypress" />
-
 import {
-  AD_PATH,
   AD_URL,
-  API_URL_PREFIX,
   APP_URL_PREFIX,
   DASHBOARD,
-  DETECTORS,
   SLASH,
 } from '../../../utils/constants';
 
 context('AD Dashboard', () => {
-  it('Empty dashboard - no detector', () => {
-    cy.server();
-    cy.route(
-      'GET',
-      [API_URL_PREFIX, AD_PATH, DETECTORS + '*'].join(SLASH),
-      'fixture:no_detector_index_response.json'
-    ).as('getDetectors');
+  it('Empty dashboard - no detector index', () => {
+    cy.mockGetDetectorOnAction('no_detector_index_response.json', () => {
+      cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+    });
+    cy.contains('h2', 'You have no detectors');
+  });
 
-    cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+  it('Empty dashboard - empty detector index', () => {
+    cy.mockGetDetectorOnAction('empty_detector_index_response.json', () => {
+      cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+    });
+    cy.contains('h2', 'You have no detectors');
+  });
 
-    cy.wait('@getDetectors', { requestTimeout: 60_000 });
+  it('AD dashboard - single stopped detector', () => {
+    cy.mockGetDetectorOnAction('single_detector_index_response.json', () => {
+      cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+    });
 
-    cy.contains('h2', 'You have no detectors');
+    cy.contains('h3', 'Live anomalies');
+    cy.contains(
+      'p',
+      'All matching detectors are under initialization or stopped for the last 30 minutes. Please adjust filters or come back later.'
+    );
+  });
+
+  it('AD dashboard - create detector', () => {
+    cy.mockGetDetectorOnAction('no_detector_index_response.json', () => {
+      cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+    });
+
+    cy.mockSearchIndexOnAction('search_index_response.json', () => {
+      cy.get('.euiButton--primary.euiButton--fill:first').click({
+        force: true,
```

Comment:
why do we have to set `force: true` for all click/type actions?

Reply:
Kibana body has a <div.content> which has `overflow: hidden`, and it is the parent of all divs in the page. As per [cypress doc](https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Visibility), such element will be considered as invisible. If `force` is not set as true, error will show up as below:

![Screen Shot 2020-06-16 at 1 03 47 PM](https://user-images.githubusercontent.com/59710443/84822653-e4362200-afd1-11ea-9d69-ea98cafe25b7.png)

Reply:
Got it, thanks